### ISSUE TITLE
yolox does not require loss creator when running on orca

### ISSUE THREAD

**Issue by zhanglirong1999** [intel-analytics/BigDL#4412]
When train yolox on orca, there is no need to pass the 'loss creator' as parameter to initialize Estimator and calculate the loss after forward(). Because yolox can directly get the loss after forward(). 
https://github.com/Megvii-BaseDetection/YOLOX/blob/main/yolox/core/trainer.py#L106

The same is true when evaluate yolox on orca, there is no need to calculate the loss after forward()

**Comment by zhanglirong1999** [comment]
Orca calculates loss after forward(When run yolox, there is no longer need to compute the loss function).
This is the link that orca calculate the loss:
https://github.com/intel-analytics/BigDL/blob/main/python/orca/src/bigdl/orca/learn/pytorch/training_operator.py#L285

**Comment by zhanglirong1999** [comment]
So can Orca change loss to be optional ?

**Comment by hkvision** [comment]
Probably we can make loss_creator optional. If so, they inputs and targets would all be fed into the model, and user should define a function to get the loss from the model forward result (and then to combine with loss.backward)?
